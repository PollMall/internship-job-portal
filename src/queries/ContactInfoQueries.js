import { gql } from '@apollo/client';

export const UPDATE_CONTACT_INFO = gql`
mutation(
  $id: Int!
  $email: String
  $phone: String
  $city: String
  $countryId: Int
  $website: String
  $avatarUrl: String
  $about: String
) {
  updateContactInfo(
    id: $id
    email: $email
    phone: $phone
    city: $city
    countryId: $countryId
    website: $website
    avatarUrl: $avatarUrl
    about: $about
  ) {
    id
    email
    phone
    city
    country {
      id
      name
    }
    website
    avatarUrl
    about
  }
}
`;

export const CREATE_CONTACT_INFO = gql`
mutation(
  $email: String!
  $phone: String!
  $city: String!
  $countryId: Int!
  $website: String
  $avatarUrl: String
  $about: String
) {
  createContactInfo(
    email: $email
    phone: $phone
    city: $city
    countryId: $countryId
    website: $website
    avatarUrl: $avatarUrl
    about: $about
  ) {
    id
    email
    phone
    city
    country {
      id
      name
    }
    website
    avatarUrl
    about
  }
}
`;
