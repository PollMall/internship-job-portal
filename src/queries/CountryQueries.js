import { gql } from '@apollo/client';

export const GET_COUNTRIES = gql`
query {
  counties {
    id
    name
  }
}
`;
