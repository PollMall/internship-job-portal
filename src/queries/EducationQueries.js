import { gql } from '@apollo/client';

export const CREATE_USER_EDUCATION = gql`
mutation(
  $institution: String
  $description: String
  $userId: Int!
  $startDate: String
  $endDate: String
) {
  createUserEducation(
    institution: $institution
    description: $description
    userId: $userId
    startDate: $startDate
    endDate: $endDate
  ) {
    id
    institution
    description
    startDate
    endDate
  }
}
`;

export const DELETE_USER_EDUCATION = gql`
mutation($id: [Int!]!) {
  deleteUserEducation(id: $id)
}
`;

export const UPDATE_USER_EDUCATION = gql`
mutation(
  $id: Int!
  $institution: String
  $description: String
  $userId: Int!
  $startDate: String
  $endDate: String
) {
  updateUserEducation(
    id: $id
    institution: $institution
    description: $description
    userId: $userId
    startDate: $startDate
    endDate: $endDate
  ) {
    id
    institution
    description
    startDate
    endDate
  }
}
`;
