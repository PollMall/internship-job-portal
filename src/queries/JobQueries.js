import { gql } from '@apollo/client';

export const GET_JOBS = gql`
query {
  jobs {
    id
    name
    description
    isAvailable
    company {
      name
    }
    updatedAt
  }
}
`;

export const GET_JOB = gql`
query($id: Int!) {
  job(id: $id) {
    id
    name
    description
    company {
      name
      user {
        firstName
        lastName
      }
      contactInfo {
        email
        phone
        city
        country {
          name
        }
        website
        avatarUrl
        about
      }
    }
    jobSkills {
      skill {
        name
      }
      rating
    }
    jobRequirements {
      name
    }
    jobBenefits {
      name
    }
  }
}
`;
