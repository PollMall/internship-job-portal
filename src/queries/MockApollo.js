/* eslint-disable import/prefer-default-export */

export const mockResponseType = {
  SUCCESS: 'SUCCESS',
  ERROR: 'ERROR',
};

export const makeMock = (mockData, type) => {
  const { query, variables, response } = mockData;

  switch (type) {
    case mockResponseType.SUCCESS:
      return {
        request: {
          query,
          variables,
        },
        result: {
          data: response,
        },
      };
    case mockResponseType.ERROR:
      return {
        request: {
          query,
          variables,
        },
        error: response,
      };
    default:
      return new Error('Mock response not supported');
  }
};
