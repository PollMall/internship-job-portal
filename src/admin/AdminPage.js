import React, { useContext } from 'react';
import { Typography } from '@material-ui/core';
import { useQuery } from '@apollo/client';
import { UserContext } from '../UserProvider';
import { GET_USERS } from '../queries/UserQueries';
import UsersTable from './UsersTable';
import useStyles from './useStyles';
import Info from '../Info';

function AdminPage({ errorAutoHide = 2500 }) {
  const classes = useStyles();
  const {
    data, loading, error, refetch,
  } = useQuery(GET_USERS);
  const { state } = useContext(UserContext);

  return (
    <>
      <Typography className={classes.pageTitle} variant="h4">
        List of users
      </Typography>
      <Info loading={loading} error={!!error} errorAutoHide={errorAutoHide} />
      {data
      && (
      <UsersTable
        users={data.users?.filter((u) => (state?.user.id !== u.id))}
        refetch={refetch}
      />
      )}
    </>
  );
}

export default AdminPage;
