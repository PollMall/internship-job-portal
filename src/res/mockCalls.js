import { users, userRoles } from './mockData';
import { GET_USERS, DELETE_USER, GET_USER_ROLES } from '../queries/UserQueries';

export const mockGetUsers = {
  query: GET_USERS,
  variables: {},
  response: {
    users,
  },
};

export const mockDeleteUser = {
  query: DELETE_USER,
  variables: { id: 1 },
  response: {
    deleteUser: true,
  },
};

export const getUserRoles = {
  query: GET_USER_ROLES,
  variables: {},
  response: {
    userRoles,
  },
};
