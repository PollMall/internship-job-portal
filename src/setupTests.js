// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';
import { render } from '@testing-library/react';
import { mockApolloData } from './queries/MockApolloData';

Object.defineProperty(window, 'matchMedia', {
  value: () => ({
    matches: false,
    addListener: () => {},
    removeListener: () => {},
  }),
});

Object.defineProperty(window, 'alert', {
  value: jest.fn()
})

function makeMockApolloCalls(props) {
  return props.map((p) => {
    const { query, variables, data} = p;
    console.log(data);
    return {
      request: {
        query,
        variables: variables || {}
      },
      result: {
        data: data || {}
      }
    }
  })
}

const mockApolloCalls = makeMockApolloCalls(mockApolloData);

const AllProviders = ({ children }) => {
  return (
    <MockedProvider mocks={mockApolloCalls} addTypename={false}>
       <MemoryRouter>
         {children}
       </MemoryRouter>
     </MockedProvider>
  )
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllProviders, ...options })


export { customRender as render };