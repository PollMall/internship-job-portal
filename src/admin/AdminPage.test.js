import React from 'react';
import { screen } from '@testing-library/react';
import { render } from '../setupTests';
import AdminPage from './AdminPage';
import { makeMock, mockResponseType } from '../queries/MockApollo';
import { mockGetUsers as mockData } from '../res/mockCalls';

describe('AdminPage tests', () => {
  it('should match snapshot', () => {
    const { container } = render(<AdminPage />, undefined, { user: {} });
    expect(container).toMatchSnapshot();
  });

  it('should display table', async () => {
    render(<AdminPage />, [makeMock(mockData, mockResponseType.SUCCESS)], { user: {} });
    expect(await screen.findByRole('table')).toBeInTheDocument();
  });
});
