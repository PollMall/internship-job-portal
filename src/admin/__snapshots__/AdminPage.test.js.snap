// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminPage tests should match snapshot 1`] = `
<div>
  <h4
    class="MuiTypography-root makeStyles-pageTitle-2 MuiTypography-h4"
  >
    List of users
  </h4>
  <div
    class="MuiCircularProgress-root makeStyles-progress-10 MuiCircularProgress-indeterminate"
    role="progressbar"
    style="width: 40px; height: 40px;"
  >
    <svg
      class="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </div>
</div>
`;
